S/SHTOOLS/build_tarballs.jl

```diff
@@ -41,12 +48,10 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency("CompilerSupportLibraries_jll"),
-    Dependency("FFTW_jll"),
-    Dependency("OpenBLAS32_jll"),
+    Dependency("FFTW_jll"; compat="3.3.11"),
+    Dependency("OpenBLAS32_jll"; compat="0.3.29"),
```

Comment:
Why do we need to add a compat entry here?

Reply:
I don't think it's necessary, but it's good style to define compatibility. Without such an entry, (I think) BinaryBuilder would build agains 0.3.29 and then set an automatic compat entry of 0.3, which then also allows earlier versions.

Is this a generic questions or do you have a concern?

Reply:
> and then set an automatic compat entry of 0.3

That's not true, there would simply be no explicit compat bound (which for openblas specifically is probably OK, as it has a fairly stable ABI)

Reply:
My concern would be a compat of 0.3.29 basically says 0.3.29+, which would basically force compatibility with only Julia 1.11 and newer (OpenBLAS32_jll 0.3.25+ is Julia 1.11+ only), so it seems pretty aggressive to raise the compat that high unless there is a reason the package needs a specific API/ABI of the dependency.

Reply:
I thought Julia included `OpenBLAS`, and we're using `OpenBLAS32` here. Aren't these two unrelated?

If not then I definitely need to remove that compat bound again.

Reply:
While we don't include it as a stdlib, we have generally bumped the Julia compat for both OpenBLAS and OpenBLAS32 at the same time (since we always update them in tandem).

Reply:
Compat bound removed.